src/Shared/LayoutRenderers/AspNetRequestPostedBodyLayoutRenderer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Text;
+using NLog.LayoutRenderers;
+#if ASP_NET_CORE
+using Microsoft.AspNetCore.Http;
+#else
+using System.Web;
+#endif
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// ASP.NET posted body, e.g. FORM or Ajax POST
+    /// </summary>
+    /// <para>Example usage of ${aspnet-request-posted-body}:</para>
+    /// <example>
+    /// <code lang="NLog Layout Renderer">
+    /// ${aspnet-request-posted-body} - Produces - {username:xyz,password:xyz}
+    /// </code>
+    /// </example>
+    [LayoutRenderer("aspnet-request-posted-body")]
+    public class AspNetRequestPostedBodyLayoutRenderer : AspNetLayoutRendererBase
+    {
+
+        /// <summary>
+        /// The string for the key in HttpContext.Items for the POST request body
+        /// </summary>
+        public static readonly string NLogPostedRequestBodyKey = "__nlog-aspnet-request-posted-body";
```

Comment:
Could this become `internal` instead of `public` ?

Reply:
Any reason why you use a string-key with slow hashcode, instead of standard object:
```c#
internal static readonly object NLogPostedRequestBodyKey = new object();
```

Reply:
Made internal.

Reply:
Converted to object as requested.